firewalls.tf

```diff
@@ -40,12 +40,35 @@ resource "google_compute_firewall" "allow_idapi_egress" {
   destination_ranges = var.idapi_cidr_ip_addresses
 }
 
+resource "google_compute_firewall" "allow_query_engine_egress" {
```

Comment:
I feel, all the egress rules can be combined into a single one, what needs to be configurable is just the destination_ranges and ports.

Said that, the conditional for the rule here, should be "enable_query_engine_egress" and not "enable_dataproc_network"; TBH, after analysing the enable_dataproc_network variable doesnt make much sense in my view.